internal/services/mssql/mssql_virtual_machine_resource.go

```diff
@@ -431,6 +432,41 @@ func resourceMsSqlVirtualMachine() *pluginsdk.Resource {
 				},
 			},
 
+			"sql_virtual_machine_group_id": {
+				Type:             pluginsdk.TypeString,
+				Optional:         true,
+				ValidateFunc:     sqlvirtualmachinegroups.ValidateSqlVirtualMachineGroupID,
+				DiffSuppressFunc: suppress.CaseDifference,
```

Comment:
the subscription id becomes upper-case after creation, I've created https://github.com/Azure/azure-rest-api-specs/issues/25211 to track the issue. Meanwhile, instead of using `suppress.CaseDifference`, I've updated the code to get the subscriptionId with correct case to manually construct this property